thirdparty/github.go

```diff
@@ -1333,24 +1329,34 @@ func MostRestrictiveGitHubPermission(perm1, perm2 string) string {
 // GetPullRequestMergeBase returns the merge base hash for the given PR.
 // This function will retry up to 5 times, regardless of error response (unless
 // error is the result of hitting an api limit)
-func GetPullRequestMergeBase(ctx context.Context, data GithubPatch) (string, error) {
+func GetPullRequestMergeBase(ctx context.Context, owner, repo, baseRevision, currentCommitHash string, prNum int) (string, error) {
+	mergeBase, err := GetGithubMergeBaseRevision(ctx, owner, repo, baseRevision, currentCommitHash)
+	if err == nil {
+		return mergeBase, nil
+	}
+	apiErr, ok := errors.Cause(err).(APIRequestError)
+	if !ok || apiErr.StatusCode != http.StatusNotFound {
+		return "", err
+	}
```

Comment:
Is there any reason to exit early on these specific error cases, rather than just retrying on any error? If the logic below ultimately behaves the same as `GetGithubMergeBaseRevision`, then getting the merge base may be more resilient to other weird/unexpected GitHub results if it falls back on any error.

Reply:
That's fair, changed to fallback in all err cases